scratch-js/Sprite.mjs

```diff
@@ -91,7 +91,35 @@ class SpriteBase {
     }
     if (typeof costume === "string") {
       const index = this.costumes.findIndex(c => c.name === costume);
-      if (index > -1) this.costumeNumber = index + 1;
+      if (index > -1) {
+        this.costumeNumber = index + 1;
+      } else {
```

Comment:
There's a lot of duplicated logic between the stage/sprite implementations.

Maybe use a `switch` statement:

```js
        switch (costume) {
          case "next costume":
          case "next backdrop": {
            this.costumeNumber++;
            break;
          }

          case "previous costume":
          case "previous backdrop": {
            this.costumeNumber--;
            break;
          }

          case "random backdrop": {
            if (this instanceof Stage) {
              const numCostumes = this.costumes.length;
              if (numCostumes > 1) {
                const currentIndex = this.costumeNumber - 1;
                let selectedIndex = Math.floor(
                  Math.random() * (numCostumes - 1)
                );
                if (selectedIndex === currentIndex) selectedIndex += 1;
                this.costumeNumber = selectedIndex + 1;
              }
            }
            break;
          }

          default: {
            if (!(isNaN(costume) || costume.trim().length === 0)) {
              this.costumeNumber = Number(costume);
            }
          }
        }
```

Reply:
I was thinking such, but it doesn't perfectly match Scratch behavior then - sprites should never respond to "next/previous backdrop". It'd be nice if JS had syntax like `case "previous backdrop" && this instanceof Stage`, buut yeah.

That said, I think it might be fine to drop such a little bit of consistency with Scratch, in favor of much more readable code - especially when you consider that "backdrops" don't really exist as a term in scratch-js. Presently, `stage.costume = "next costume"` just wouldn't work, even though it's the most natural code. So- I'll update the code, thanks!

Reply:
Alright, [pushed](https://github.com/PullJosh/scratch-js/pull/45/commits/035c741651d4ef8be4634bf24946ffc95faa50a7). Thoughts on the above? (Thanks for the review btw!!)

Reply:
Looks good!